src/test/java/org/takes/facets/fork/FkAgentTest.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.facets.fork;
+
+import java.io.IOException;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.takes.facets.fork.am.AmVersion;
+import org.takes.rq.RqFake;
+import org.takes.rq.RqWithHeader;
+import org.takes.tk.TkEmpty;
+
+/**
+ * Test case for {@link FkAgent}.
+ * @author Valeriy Zhirnov (neonailol@gmail.com)
+ * @version $Id$
+ * @since 1.7.2
+ */
+public final class FkAgentTest {
+
+    /**
+     * Some small version number.
+     */
+    private static final int VERSION_12 = 12;
+
+    /**
+     * Some big version number.
+     */
+    private static final int VERSION_90 = 90;
+
+    /**
+     * FkAgent can match by User-Agent header.
```

Comment:
@neonailol this javadoc is too abstract, it's OK to remove javadocs in test methods and add `@checkstyle JavadocMethodCheck (500 lines)` suppression to test class (if you have good test method name, like in this case)